src/attachSpellCheckProvider.ts

```diff
@@ -23,7 +23,7 @@ interface ProviderProxy {
   /**
    * Add a word to the current dictionary.
    */
-  addWord: (text: string) => Promise<void>;
+  addWord?: (languageKey: string, text: string) => Promise<void>;
```

Comment:
I'd leave this as required. Since providerproxy requires implementation to `spellCheckerProvider` anyway, it'd better to provide full interfaces instead of existence checking. If providerProxy implemented via es proxy (something like `comlink`) it's hard to check object property exists.

Reply:
That would involve making this a breaking change.. you ok with that?

Reply:
Uh you're right :thinking:

Reply:
Ok, let's leave it as optional and I can make required for next major breaking later.